Can not determin type of display interface

- Version of `ssd1306` in use (if applicable): Master branch of github.
- MCU/other hardware in use: STM32f103 the BluePill
- Display resolution and interface: [I2C], [128x64]

## Description of the problem/feature request/other

Can not determin type of display interface, I want to use display in critical sections, But not working, what can I do about it ,Thanks!

......
```rust
type DISP = ssd1306::mode::graphics::GraphicsMode<
     stm32f1xx_hal::i2c::BlockingI2c<
         stm32f1::stm32f103::I2C1,
         (
             stm32f1xx_hal::gpio::gpiob::PB8<stm32f1xx_hal::gpio::Alternate<OpenDrain>>,
             stm32f1xx_hal::gpio::gpiob::PB9<stm32f1xx_hal::gpio::Alternate<OpenDrain>>,
         ),
     >,
 >;
```
......

```rust

let i2c = BlockingI2c::i2c1(
        device.I2C1,
        (scl, sda),
        &mut afio.mapr,
        Mode::Fast {
            frequency: 400_000.hz(),
            duty_cycle: DutyCycle::Ratio2to1,
        },
        clocks,
        &mut rcc.apb1,
        1000,
        10,
        1000,
        1000,
    );
let interface = I2CDIBuilder::new().init(i2c);

// The working code.
// let mut disp: GraphicsMode<_> = Builder::new().connect(interface).into();

// **Code Not Working.
let mut disp: DISP = Builder::new().connect(interface).into();
```

## Compile appears errors.

```shell
error[E0277]: the trait bound `ssd1306::mode::graphics::GraphicsMode<stm32f1xx_hal::i2c::BlockingI2c<stm32f1::stm32f103::I2C1, (stm32f1xx_hal::gpio::gpiob::PB8<stm32f1xx_hal::gpio::Alternate<stm32f1xx_hal::gpio::OpenDrain>>, stm32f1xx_hal::gpio::gpiob::PB9<stm32f1xx_hal::gpio::Alternate<stm32f1xx_hal::gpio::OpenDrain>>)>>: ssd1306::mode::displaymode::DisplayModeTrait<impl display_interface::WriteOnlyDataCommand, ssd1306::displaysize::DisplaySize128x64>` is not satisfied
   --> src\main.rs:183:60
    |
183 |     let mut disp: DISP = Builder::new().connect(interface).into();
    |                                                            ^^^^ the trait `ssd1306::mode::displaymode::DisplayModeTrait<impl display_interface::WriteOnlyDataCommand, ssd1306::displaysize::DisplaySize128x64>` is not implemented for `ssd1306::mode::graphics::GraphicsMode<stm32f1xx_hal::i2c::BlockingI2c<stm32f1::stm32f103::I2C1, (stm32f1xx_hal::gpio::gpiob::PB8<stm32f1xx_hal::gpio::Alternate<stm32f1xx_hal::gpio::OpenDrain>>, stm32f1xx_hal::gpio::gpiob::PB9<stm32f1xx_hal::gpio::Alternate<stm32f1xx_hal::gpio::OpenDrain>>)>>`
    |
    = help: the following implementations were found:
              <ssd1306::mode::graphics::GraphicsMode<DI, DSIZE> as ssd1306::mode::displaymode::DisplayModeTrait<DI, DSIZE>>
```

--- a/src/builder.rs
+++ b/src/builder.rs
@@ -160,7 +160,10 @@ impl I2CDIBuilder {
     /// Finish the builder and return an initialised display interface for further use
     ///
     /// This method consumes the builder and must come last in the method call chain
-    pub fn init<I: hal::blocking::i2c::Write>(self, i2c: I) -> impl WriteOnlyDataCommand {
+    pub fn init<I: hal::blocking::i2c::Write>(
+        self,
+        i2c: I,
+    ) -> display_interface_i2c::I2CInterface<I> {
         display_interface_i2c::I2CInterface::new(i2c, self.i2c_addr, 0x40)
     }
 }

